src/modules/map/components/CompareSlider.vue

```diff
@@ -10,27 +10,65 @@ const props = defineProps({
         default: window.innerWidth,
     },
 })
-
+const getMap = inject('getMap')
 const { clientWidth } = toRefs(props)
+const olMap = ref(null)
 
 const compareRatio = ref(-0.5)
-
 const store = useStore()
 const storeCompareRatio = computed(() => store.state.ui.compareRatio)
 const compareSliderPosition = computed(() => {
     return {
         left: compareRatio.value * 100 + '%',
     }
 })
+const visibleLayerOnTop = computed(() => store.getters.visibleLayerOnTop)
 
 watch(storeCompareRatio, (newValue) => {
     compareRatio.value = newValue
 })
 
+watch(visibleLayerOnTop, () => {
+    nextTick(slice())
+})
+
 onMounted(() => {
     compareRatio.value = storeCompareRatio.value
+    slice()
 })
 
+function slice() {
+    if (olMap.value) {
+        olMap.value.un('prerender', onPreCompose)
+        olMap.value.un('postrender', onPostCompose)
+    }
+    olMap.value = getMap()
```

Comment:
As said above, no storing of big object such as the map in a ref, that is a huge performance issue